### PR TITLE
[API Documentation] Add basic documentation for chrome.action

### LINKED ISSUE #193: Enhance the action API reference docs
Once the generated docs for the action API are included, need to update this by:

- Add enhanced introductory text to provide context and more how-to focus
- Add xrefs to the new API reference from places that mention use of the action API.

### PR DESCRIPTION
**Pull request by rdcronin**
Add basic documentation for the MV3 chrome.action API, including
descriptions of the different components of the UI and a handful
of (very) small examples.

Fixes #193

### CODE REVIEW

File: site/en/docs/extensions/reference/action/index.md [hunk 1]
```diff
@@ -2,9 +2,165 @@
...
 ---
 
-!!!.aside.warning
-This documentation is in active development.
-We're planning to expand the introduction section to provide more context and
-guidance on how to use this API.  This work is tracked in issue
-[#193](https://github.com/GoogleChrome/developer.chrome.com/issues/193).
+You can use the `chrome.action` API to control the toolbar button for your
+extension in Chrome's UI.  The action icons are displayed in the browser
+toolbar, to the right of the omnibox (on LTR devices). After installation, by
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:7]: Does LTR mean left-to-right? If so, suggestion is "on left-to-right devices".

File: site/en/docs/extensions/reference/action/index.md [hunk 2]
```diff
@@ -2,9 +2,165 @@
...
+default, these will be in the extensions menu (the puzzle piece). Users can
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:8]: will be -> appear

File: site/en/docs/extensions/reference/action/index.md [hunk 3]
```diff
@@ -2,9 +2,165 @@
...
+For packed extensions (installed from a .crx file), images can be in most
+formats that the Blink rendering engine can display, including PNG, JPEG,
+BMP, ICO, and others (SVG is not supported). Unpacked extensions must use
+images in the PNG format.
+
+### Tooltip (title)
+
+The tooltip, or title, is displayed when the user hovers the mouse of the
+extension's icon in the toolbar. It is also included in the accessible text
+spoken by screenreaders when the button is focused.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:89]: "is focused"? Consider "gets focus"

File: site/en/docs/extensions/reference/action/index.md [hunk 4]
```diff
@@ -2,9 +2,165 @@
...
+#### Formats
+
+For packed extensions (installed from a .crx file), images can be in most
+formats that the Blink rendering engine can display, including PNG, JPEG,
+BMP, ICO, and others (SVG is not supported). Unpacked extensions must use
+images in the PNG format.
+
+### Tooltip (title)
+
+The tooltip, or title, is displayed when the user hovers the mouse of the
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:87]: Suggestion:
...or title, appears when...

File: site/en/docs/extensions/reference/action/index.md [hunk 5]
```diff
@@ -2,9 +2,165 @@
...
+extension's icon in the toolbar. It is also included in the accessible text
+spoken by screenreaders when the button is focused.
+
+The default tooltip is set from the `default_title` field in the `action`
+`manifest.json` file. It can also be set programmatically with the
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:92]: Suggestion:
...you can also set it programmatically using the...

File: site/en/docs/extensions/reference/action/index.md [hunk 6]
```diff
@@ -2,9 +2,165 @@
...
+    () => { ... });
+
+chrome.action.setBadgeBackgroundColor(
+    {color: '#00FF00'}  // Also green
+    () => { ... });
+```
+
+### Popup
+
+An action's popup will be shown when the user clicks on the extension's button
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:123]: ...popup appears when the user clicks...

File: site/en/docs/extensions/reference/action/index.md [hunk 7]
```diff
@@ -2,9 +2,165 @@
...
+in the toolbar. The popup can contain any HTML contents you like, and will be
+sized to fit its contents (though it has a maximum size).
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:125]: ...and is sized to fit...

File: site/en/docs/extensions/reference/action/index.md [hunk 8]
```diff
@@ -2,9 +2,165 @@
...
+
+The popup is initially set from the `default_popup` property in the `action`
+key in the `manifest.json` file. If present, this should point to a relative
+path within the extension directory. It can also be updated dynamically to
+point to a different relative path using the `action.setPopup()` method.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:130]: You can also update it dynamically...

File: site/en/docs/extensions/reference/action/index.md [hunk 9]
```diff
@@ -2,9 +2,165 @@
...
+
+!!!.aside
+The `action.onClicked` event will not be dispatched if the extension action
+has a popup on the current tab.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:134]: Maybe "already has a popup" makes this clearer, if that is indeed the case.

File: site/en/docs/extensions/reference/action/index.md [hunk 10]
```diff
@@ -2,9 +2,165 @@
...
 !!!
+
+## Per-tab state
+
+Extension actions can have different states for each tab. For instance, you
+could set the badge text to be different on each tab (to show tab-specific state).
+You can set the value for an individual tab using the `tabId` property in the
+different setting methods on the `action` API. For instance, to set the badge
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:142]: Hmmm. "different setting methods" reads wrong. Consider one of:

"...the various setting methods on..."
"...each of the setting methods on..."

File: site/en/docs/extensions/reference/action/index.md [hunk 11]
```diff
@@ -2,9 +2,165 @@
...
+```js
+const tabId = getTabId();
+const tabMessage = getTabMessage(tabId);
+chrome.action.setBadgeText(
+    {text: tabMessage, tabId: tabId},
+    () => { ... });
+```
+
+If the `tabId` property is omitted, the setting is treated as a global setting.
+Tab-specific settings will take priority over any global settings.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:154]: ...settings take priority over...

https://developers.google.com/style/tense

To elaborate on the rationale: saying "a takes priority over b" eliminates the fuzziness of "a will take priority over b", which can imply that it happens at some later point. One less bump in the road for the reader.

File: site/en/docs/extensions/reference/action/index.md [hunk 12]
```diff
@@ -2,9 +2,165 @@
...
+
+## Enabled state
+
+By default, toolbar actions are enabled (clickable) on every tab. You can
+control this by using the `action.enable()` and `action.disable()` methods.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:159]: "...control this using the..."

File: site/en/docs/extensions/reference/action/index.md [hunk 13]
```diff
@@ -2,9 +2,165 @@
...
+This will only affect whether the popup (if any) or `action.onClicked` event
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/action/index.md:160]: "This only affects..."

**Review by awfuchs — CHANGES_REQUESTED** [review]

File: site/en/docs/extensions/reference/action/index.md (hunk 1 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:7]: Yep, Left-to-Right (as opposed to an RTL language, like Arabic, where much of the UI is reversed).  Updated.

File: site/en/docs/extensions/reference/action/index.md (hunk 2 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:8]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 3 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:89]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 4 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:87]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 5 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:92]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 9 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:134]: Not quite - it's not if the popup is showing; it's if the popup is specified at all.   onClick is only dispatched if no popup is specified for the tab.

I've rephrased to "has specified a popup to show on click on the current tab.", which might be more clear.  Lemme know what you think.

File: site/en/docs/extensions/reference/action/index.md (hunk 10 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:142]: I like "the various setting".  Done

File: site/en/docs/extensions/reference/action/index.md (hunk 11 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:154]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 12 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:159]: Done

File: site/en/docs/extensions/reference/action/index.md (hunk 13 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/action/index.md:160]: Done

**Review by awfuchs — APPROVED** [review]

**Pull request merged**